source/material_model/grain_size.cc

```diff
@@ -1313,6 +1364,14 @@ namespace aspect
                  ExcMessage("The parameter 'Use paleowattmeter' has been removed. "
                             "Use the parameter 'Grain size evolution formulation instead'."));
 
+          volume_fraction_phase_one             = Utilities::string_to_double
+                                                  (Utilities::split_string_list(prm.get ("Volume fraction of phase")));
+
+          if (volume_fraction_phase_one.size()>1)
+            for (unsigned int i=0; i<volume_fraction_phase_one.size(); ++i)
+              AssertThrow( (volume_fraction_phase_one[i] == 0. || volume_fraction_phase_one[i] == 1.),
+                           ExcMessage("Error: Volume fraction should be between (0, 1) to use two phase flow !"));
```

Comment:
I am confused by this code. What is it supposed to do? Did you test that this works as intended? 
To me, it looks like now you can only pick 0 or 1 as volume fractions...

I think what you want to do might be this:
```suggestion
          if (volume_fraction_phase_one.size()>1)
            for (unsigned int i=0; i<volume_fraction_phase_one.size(); ++i)
              AssertThrow( (volume_fraction_phase_one[i] != 0. && volume_fraction_phase_one[i] != 1.),
                           ExcMessage("Error: Volume fraction must be between (0, 1) to use two phase damage!"));
```

Reply:
Yes, this is what I wanted to do. Thank you!